### PR TITLE
(FM-7333) - String-ify

### PR DESCRIPTION
**Pull request by lionce**
All Puppet files are documented 

- Puppet Functions:         1 (    0 undocumented)
- Puppet Plans:                 0 (    0 undocumented)
- Puppet Tasks:                 0 (    0 undocumented)
- Puppet Classes:             0 (    0 undocumented)
- Puppet Defined Types:  0 (    0 undocumented)
- Puppet Types:                2 (    0 undocumented)
- Puppet Providers:           2 (    0 undocumented)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -291,6 +291,7 @@ ini_setting { '[section1] setting3':
 
 
 ## Reference
+See [REFERENCE.md](https://github.com/puppetlabs/puppetlabs-inifile/blob/master/REFERENCE.md)
```
Comment (eimlav) [inline README.md]: Remove content of reference section as this is now in the REFERENCE.md

File: REFERENCE.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+The following parameters are available in the `ini_subsetting` type.
+
+##### `name`
+
+namevar
+
+An arbitrary name used as the identity of the resource.
+
+##### `section`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for section.

File: REFERENCE.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+##### `value`
+
+The value of the subsetting to be defined.
+
+#### Parameters
+
+The following parameters are available in the `ini_subsetting` type.
+
+##### `name`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for name.

File: REFERENCE.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+##### `show_diff`
+
+Valid values: `true`, md5, `false`
+
+Whether to display differences when the setting changes.
+
+Default value: `true`
+
+##### `key_val_separator`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for key_val_separator.

File: REFERENCE.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+**Functions**
+
+* [`create_ini_settings`](#create_ini_settings): Uses create_resources to create a set of ini_setting resources from a hash:      $settings = {  section1 => {         setting1 => val1       
+
+## Resource types
+
+### ini_setting
+
+An arbitrary name used as the identity of the resource.
```
Comment (eimlav) [inline REFERENCE.md]: This is the description for the `name` param. You will need to add a general description for ini_setting

File: REFERENCE.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+Valid values: `true`, `false`, yes, no
+
+Create setting only if the section exists
+
+Default value: `true`
+
+##### `path`
+
+The ini file Puppet will ensure contains the specified setting.
```
Comment (eimlav) [inline REFERENCE.md]: Might be worth adding that path must be absolute.

File: REFERENCE.md [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+##### `show_diff`
+
+Valid values: `true`, md5, `false`
+
+Whether to display differences when the setting changes.
+
+Default value: `true`
+
+##### `key_val_separator`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for key_val_separator.

File: REFERENCE.md [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+
+
+Default value:  = 
+
+##### `section_prefix`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for section_prefix.

File: REFERENCE.md [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+
+
+Default value: [
+
+##### `section_suffix`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for section_suffix.

File: REFERENCE.md [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+
+
+Default value: ]
+
+##### `indent_char`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for indent_char.

File: REFERENCE.md [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+
+
+Default value:  
+
+##### `indent_width`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for indent_width.

File: REFERENCE.md [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+
+
+##### `refreshonly`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for refreshonly.

File: REFERENCE.md [hunk 13]
```diff
@@ -0,0 +1,310 @@
...
+##### `subsetting_separator`
+
+The separator string between subsettings. Defaults to " "
+
+Default value:  
+
+##### `subsetting_key_val_separator`
+
+The separator string between the subsetting name and its value. Defaults to the empty string.
```
Comment (eimlav) [inline REFERENCE.md]: Remove `Defaults to...`

File: REFERENCE.md [hunk 14]
```diff
@@ -0,0 +1,310 @@
...
+
+Default value: ''
+
+##### `path`
+
+The ini file Puppet will ensure contains the specified setting.
```
Comment (eimlav) [inline REFERENCE.md]: Mention the need for an absolute path.

File: REFERENCE.md [hunk 15]
```diff
@@ -0,0 +1,310 @@
...
+Default value: `true`
+
+##### `key_val_separator`
+
+
+
+Default value:  = 
+
+##### `quote_char`
+
```
Comment (eimlav) [inline REFERENCE.md]: No description for quote_char.

File: REFERENCE.md [hunk 16]
```diff
@@ -0,0 +1,310 @@
...
+Set to true if your subsettings don\'t have values and you want to use exact matches to determine if the subsetting exists. See MODULES-2212
+
+Default value: `false`
+
+##### `insert_type`
+
+Valid values: start, end, before, after, index
+
+Where the new subsetting item should be inserted?
```
Comment (eimlav) [inline REFERENCE.md]: Remove `?`.

File: REFERENCE.md [hunk 17]
```diff
@@ -0,0 +1,310 @@
...
+      path    => '/tmp/foo.ini',
+    }
+    ini_setting{'/tmp/foo.ini [section2] setting3':
+      ensure  => absent,
+      section => 'section2',
+      setting => 'setting3',
+      path    => '/tmp/foo.ini',
+    }
+
+#### `create_ini_settings()`
```
Comment (eimlav) [inline REFERENCE.md]: Duplicated examples below.

File: REFERENCE.md [hunk 18]
```diff
@@ -0,0 +1,310 @@
...
+
+Default value: end
+
+##### `insert_value`
+
+The value for the insert types which require one.
+
+## Functions
+
+### create_ini_settings
```
Comment (eimlav) [inline REFERENCE.md]: Function not using Puppet Strings syntax. https://github.com/puppetlabs/puppetlabs-accounts/blob/master/lib/puppet/functions/accounts_ssh_options_parser.rb might help to figure out how to do this.

File: README.md [hunk 19]
```diff
@@ -291,6 +291,7 @@ ini_setting { '[section1] setting3':
...
 
 ### Public Types
```
Comment (eimlav) [inline README.md]: Reference documentation needs removed as it is now present in REFERENCE.md. Just leave the header and the `See REFERENCE.md` link.

**Comment by pmcmaw** [comment]
I have checked out your PR and when running `puppet strings generate' I get different output than what is posted in the PR. 

```
Files:                    9
Modules:                  2 (    1 undocumented)
Classes:                  4 (    0 undocumented)
Constants:                0 (    0 undocumented)
Attributes:               6 (    0 undocumented)
Methods:                 34 (   12 undocumented)
Puppet Classes:           0 (    0 undocumented)
Puppet Defined Types:     0 (    0 undocumented)
Puppet Types:             2 (    0 undocumented)
Puppet Functions:         1 (    0 undocumented)
Puppet Providers:         2 (    0 undocumented)
Puppet Tasks:             0 (    0 undocumented)
Puppet Plans:             0 (    0 undocumented)
 74.51% documented
```
It seems there are some methods and 1 module still undocumented.

**Pull request merged**